src/Core/Services/Implementations/PostalMailDeliveryService.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Threading.Tasks;
+using Bit.Core.Settings;
+using Microsoft.Extensions.Logging;
+using System.Net.Http;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Microsoft.AspNetCore.Hosting;
+using System.Text;
+
+namespace Bit.Core.Services
+{
+    public class PostalMailDeliveryService : IMailDeliveryService
+    {
+        private readonly GlobalSettings _globalSettings;
+        private readonly ILogger<PostalMailDeliveryService> _logger;
+        private readonly IHttpClientFactory _clientFactory;
+        private readonly string _baseTag;
+        private readonly string _from;
+        private readonly string _reply;
+
+        public PostalMailDeliveryService(
+            GlobalSettings globalSettings,
+            ILogger<PostalMailDeliveryService> logger,
+            IWebHostEnvironment hostingEnvironment,
+            IHttpClientFactory clientFactory)
+        {
+            _globalSettings = globalSettings;
+            _logger = logger;
+            _clientFactory = clientFactory;
+            _baseTag = $"Env_{hostingEnvironment.EnvironmentName}-" +
+                $"Server_{globalSettings.ProjectName?.Replace(' ', '_')}";
+            _from = $"\"{globalSettings.SiteName}\" <no-reply@{_globalSettings.Mail.PostalDomain}>";
+            _reply = $"\"{globalSettings.SiteName}\" <{globalSettings.Mail.ReplyToEmail}>";
+        }
+
+        public async Task SendEmailAsync(Models.Mail.MailMessage message)
+        {
+            var httpClient = _clientFactory.CreateClient("PostalMailDeliveryService");
+            httpClient.DefaultRequestHeaders.Add("X-Server-API-Key", _globalSettings.Mail.PostalApiKey);
+
+            var request = new PostalRequest
+            {
+                subject = message.Subject,
+                from = _from,
+                reply_to = _reply,
+                html_body = message.HtmlContent,
+                to = new List<string>(),
+                tag = _baseTag
+            };
+            foreach (var address in message.ToEmails)
+            {
+                request.to.Add(address);
+            }
+
```

Comment:
setting of `request.cc` appears to be missing.

Reply:
Whoops

Reply:
Looks like we dont support CC in our mail message model. So not needed.